crates/weaver_live_check/src/attribute_advice.rs

```diff
@@ -0,0 +1,302 @@
+        match (
+            semconv_attribute,
+            attribute.value.as_ref(),
+            attribute.r#type.as_ref(),
+        ) {
+            (Some(semconv_attribute), Some(attribute_value), Some(attribute_type)) => {
+                if let AttributeType::Enum { members, .. } = &semconv_attribute.r#type {
+                    let mut is_found = false;
+                    for member in members {
+                        if match attribute_type {
+                            PrimitiveOrArrayTypeSpec::Int => {
+                                if let Some(int_value) = attribute_value.as_i64() {
+                                    member.value == ValueSpec::Int(int_value)
+                                } else {
+                                    false
+                                }
+                            }
+                            PrimitiveOrArrayTypeSpec::String => {
+                                if let Some(string_value) = attribute_value.as_str() {
+                                    member.value == ValueSpec::String(string_value.to_owned())
+                                } else {
+                                    false
+                                }
+                            }
+                            _ => {
+                                // Any other type is not supported - the TypeAdvisor should have already caught this
+                                return Ok(Vec::new());
+                            }
+                        } {
+                            is_found = true;
+                            break;
+                        }
+                    }
+
+                    if !is_found {
+                        return Ok(vec![Advice {
+                            key: "undefined_enum_variant".to_owned(),
+                            value: attribute_value.clone(),
+                            message: "Is not a defined variant".to_owned(),
+                            advisory: Advisory::Information,
+                        }]);
+                    }
+                }
+                Ok(Vec::new())
+            }
+            _ => Ok(Vec::new()),
+        }
+    }
+}
+
+/// An advisor which runs a rego policy on the attribute
+pub struct RegoAdvisor {
+    engine: Engine,
+}
+impl RegoAdvisor {
+    /// Create a new RegoAdvisor
+    pub fn new(
+        live_checker: &AttributeLiveChecker,
+        policy_dir: &Option<PathBuf>,
+        jq_preprocessor: &Option<PathBuf>,
+    ) -> Result<Self, Error> {
+        let mut engine = Engine::new();
+        if let Some(path) = policy_dir {
+            let _ = engine
+                .add_policies(path, "*.rego")
+                .map_err(|e| Error::AdviceError {
+                    error: e.to_string(),
+                })?;
+        } else {
+            let _ = engine
+                .add_policy(
+                    "defaults/policies/advice/otel.rego",
```

Comment:
Can we move this in a constant somewhere *or* a method where it's obvious what we're adding by default?

If this isn't just pulling out all the defaults "by fiat" magically, we should very clearly have a method or location where we update this.